### ISSUE TITLE
100天机器学习->Day01数据预处理

### ISSUE THREAD

**Issue by Sunalways** [Sunalways/Blog#4]
# 100天机器学习
@author: Darcy
2019/1/31 23:23:37 

## Day01数据预处理
1. 导入需要的库
	- NumPy 数学计算函数
	- Pandas 导入和管理数据集
2.  导入数据集
	- Pandas.read_csv方法读取本地csv文件
3. 处理丢失数据
	- 处理数据可以用户整列的平均值或中间值替换丢失的数据
	- sklearn.preprocessing中Imputer完成
4. 解析分类数据
	- 分类数据含有标签值而不是数字值。需要解析成数字
	- sklearn.preprocesing导入LabelEncoder类
5. 拆分数据集为测试集合和训练集合
	- 训练集：训练模型
	- 测试集：验证模型的测试集合
	- sklearn.crossvalidation中train_test_split()方法
6. 特征缩放
	- 距离计算中，高维的特征比低维的特征权重更大。可用特征标准或Z值归一化解决
	- sklearn.preprocessing库的StandardScalar类
	
## 代码解析
1. Panda 读取数据列
	1. loc_通过行标签索引数据
		- loc[[起始行]：[终止行],[列索引]]
	2. iloc_通过行号获取行数据
		- iloc[[起始列]:[终止列],[行索引]]
	3. 结合前两种的混合索引
2. sklearn.preprocessing.Imputer类处理缺失值
	- Imputer(missing_values='NaN'(处理的缺失指)，strategy='mean'（如何处理）,axis=0(表示按列进行))
3. sklearn.preprocessing.LabelEncoder
	- 将标签分配一个0-n_classes-1之间的编码
	- labelencoder_X = LabelEncoder()
	- X[ : , 0] = labelencoder_X.fit_transform(X[ : , 0])
4. sklearn.preprocessing.OneHotEncoder独热编码	
	- 独热编号，使得数据从有序变为离散
	- onehotencoder = OneHotEncoder(categorical_features = [0])
	- X = onehotencoder.fit_transform(X).toarray()
5. sklearn.model_selection.train_test_split 拆分数据集
	- train_test_split(X, Y, test_size=0.2(划分训练集),rangdom_state=0(随机因子))
6. sklearn.preprocessing.StandardScaler特征量化
	- tandardScaler()以保存训练集中的参数（均值、方差）直接使用其对象转换测试集数据。
